### PR TITLE
Wkd selects usable key from all received

### LINKED ISSUE #3017: WKD evaluate all received keys
Currently, our client implementation only looks at the first received key.

It should at least look at all of them (WKD response may contain any amount of keys in it), and pick the first key that is usable for encryption. Spec https://tools.ietf.org/html/draft-koch-openpgp-webkey-service-10

Recommend to do after #3003

Later on we will follow up with #3018

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
This PR selects a usable key among all received from Wkd

close #3017

### CODE REVIEW

**Comment by rrrooommmaaa** [comment]
@tomholub Looks like `human@flowcrypt.com` fails due to a problem unrelated to this PR.
https://openpgpkey.flowcrypt.com/.well-known/openpgpkey/flowcrypt.com/policy is not reachable.
As for revoked keys -- our goal might be not only to skip them from Wkd's response, but to update ContactStore so these keys won't be usable. How do you think about it?

**Comment by tomholub** [comment]
>  As for revoked keys -- our goal might be not only to skip them from Wkd's response, but to update ContactStore so these keys won't be usable. How do you think about it?

Definitely - but maybe not in this PR. There is another issue #3018 to look into that as a followup.

That is also related to issue https://github.com/FlowCrypt/flowcrypt-browser/issues/3332

I think the overall sequence could be:
 - this PR
 - issue #3332 - multiple pubkeys per recipient - in this issue we would also add logic to select the right key to use if one is revoket, one is expired, one is valid etc
 - issue #3018 - also update revocations

Right now, since there is only one public key per recipient email, we'd have to create some logic that would be quite different (I think) from the final situation. Right now, let's say the recipient rotates the key. Old one is revoked, new one is available. We would have to throw out the old (revoked) one and replace it with the new one. As opposed to, after we support several pubkeys per recipient, we can independently update the revoked ones and also all of the valid ones that WKD returns - save them all. Then only once we try to email that recipient (or verify their emails) will we choose the right public key from all of the ones that are available.

**Comment by tomholub** [comment]
> @tomholub Looks like human@flowcrypt.com fails due to a problem unrelated to this PR. https://openpgpkey.flowcrypt.com/.well-known/openpgpkey/flowcrypt.com/policy is not reachable.

Sorry, fixed

**Comment by rrrooommmaaa** [comment]
> I think the overall sequence could be:
> 
> * this PR
> * issue #3332 - multiple pubkeys per recipient - in this issue we would also add logic to select the right key to use if one is revoket, one is expired, one is valid etc
> * issue #3018 - also update revocations

Ok, so this PR is ready to merge, as I understood.

File: test/source/mock/wkd/wkd-endpoints.ts [hunk 1]
```diff
@@ -116,6 +206,9 @@ export const mockWkdEndpoints: HandlersDefinition = {
   '/.well-known/openpgpkey/localhost/hu/pob4adi8roqdsmtmxikx68pi6ij35oca?l=incorrect': async () => {
     return alice; // advanced for incorrect@localhost
   },
+  '/.well-known/openpgpkey/localhost/hu/66iu18j7mk6hod4wqzf6qd37u6wejx4y?l=some.revoked': async () => {
+    return validAmongRevoked;
```
Comment (tomholub) [inline test/source/mock/wkd/wkd-endpoints.ts:210]: I just noticed - we are returning armored keys here on the WKD endpoints but that does not reflect how WKD returns keys (which should be unarmored bytes, concatenated). I'll make another issue for this.

**Review by tomholub — APPROVED** [review]
Looks good!

**Pull request merged**